### PR TITLE
GraphQL: add queries for getting pipelines

### PR DESCRIPTION
**Pull request by JeffreyThiessen**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

Adds the ability to query pipelines via graphql

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Test that all pipelines can be queried.

```graphql
query getPipelines {
	pipelines(workflowType: "available"){
    automatable
    description
    engine
    engineVersion
    executable
    name
    type
    typeVersion
    url
    version
    # workflowParams
  }
}
```

2. Test that a specific pipeline can be queried.

```graphql
query getPipeline {
	pipeline(workflowName:"phac-nml/iridanextexample",workflowVersion:"1.0.3"){
    automatable
    description
    engine
    engineVersion
    executable
    metadata
    name
    type
    typeVersion
    url
    version
    workflowParams
  }
}
```

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by ChrisHuynh333 — DISMISSED** [review]
Looks good!

**Review by ericenns — CHANGES_REQUESTED** [review]
First pass of comments

File: app/graphql/types/pipeline_type.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Types
+  # User Type
+  class PipelineType < Types::BaseType
```
[the remark this answers is not in the document]
Comment (JeffreyThiessen) [reply app/graphql/types/pipeline_type.rb]: fixed in https://github.com/phac-nml/irida-next/pull/830/commits/416ff3d7a5bcd374217c41c9a610187f074efba3

File: app/graphql/resolvers/pipelines_resolver.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Resolvers
+  # Project Resolver
+  class PipelinesResolver < BaseResolver
+    type Types::PipelineType.connection_type, null: true
```
[the remark this answers is not in the document]
Comment (JeffreyThiessen) [reply app/graphql/resolvers/pipelines_resolver.rb]: fixed in https://github.com/phac-nml/irida-next/pull/830/commits/416ff3d7a5bcd374217c41c9a610187f074efba3

File: app/graphql/types/pipeline_type.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    implements GraphQL::Types::Relay::Node
```
[the remark this answers is not in the document]
Comment (JeffreyThiessen) [reply app/graphql/types/pipeline_type.rb]: fixed in https://github.com/phac-nml/irida-next/pull/830/commits/416ff3d7a5bcd374217c41c9a610187f074efba3

File: app/graphql/types/query_type.rb [hunk 4]
```diff
@@ -41,6 +41,12 @@ class QueryType < Types::BaseObject
     field :is_puid, Boolean, null: false, resolver: Resolvers::IsPuidResolver,
                              description: 'Check if id is in puid format'
 
+    field :pipeline, Types::PipelineType, null: true, resolver: Resolvers::PipelineResolver,
+                                          description: 'Find Pipelines'
+
+    field :pipelines, Types::PipelineType.connection_type, null: true, resolver: Resolvers::PipelinesResolver,
```
[the remark this answers is not in the document]
Comment (JeffreyThiessen) [reply app/graphql/types/query_type.rb]: fixed in https://github.com/phac-nml/irida-next/pull/830/commits/416ff3d7a5bcd374217c41c9a610187f074efba3

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**